src/mongo/client/dbclient.cpp

```diff
@@ -1416,6 +1424,10 @@ bool DBClientWithCommands::eval(const string& dbname,
 }
 
 bool DBClientWithCommands::eval(const string& dbname, const string& jscode) {
```

Comment:
In general I agree with you.

In this case though you are just creating a new symbol when there doesn't really need to be one. If inlined, the compiler will just throw away the tagged one and replace with calls to the xDeprecated.

So, you are making the library one symbol bigger for no real reason.